### DISCUSSION TITLE
What happens if an empty metric set is returned?

### DISCUSSION THREAD

**Discussion by dmiyamasu** [kedacore/keda#4573] [Q&A / Need Help]
What does Keda do when a Scaler returns an empty set of metrics?
e.g. if New relic API returns an non-200 HTTP response or the request times out, how will KEDA respond?

**Comment by dmiyamasu** [comment]
reference this line of code: https://github.com/kedacore/keda/blob/main/pkg/scalers/newrelic_scaler.go#LL173

**Reply by JorTurFer** [reply]
In that case, the scaler is failing. What will happen depends on the other things in the ScaledObject.
- If this is the only trigger and there isn't a fallback configuration --> KEDA will respond with an error to tha HPA Controller and you will see the error in the HPA
- If this is the only trigger and there is a fallback configuration --> KEDA will respond with an error to the tha HPA Controller until the fallback is activated (seeing the errors in the HPA) and then, after the fallback activation, the HPA Controller will receive the fallback value
- If this isn't the only trigger and the other trigger works --> This trigger will respond as `If this is the only trigger and there is a fallback configuration` case, but if the other trigger requires more instances, the HPA controller will scale to them
- If this isn't the only trigger and the other trigger doesn't work --> Same as `If this is the only trigger and there is a fallback configuration` case
